components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -230,7 +230,7 @@ public OAuth2ClientValidationResponseDTO validateClientInfo(String clientId, Str
      */
     private boolean validateCallbackURI(String callbackURI, OAuthAppDO oauthApp) {
         String regexp = null;
-        String registeredCallbackUrl = oauthApp.getCallbackUrl();
+        String registeredCallbackUrl = OAuth2Util.processURI(oauthApp.getCallbackUrl());
```

Comment:
The callback URI is usually a specific endpoint to an app. Do we need to resolve placeholders for this?

Reply:
> @ayshsandu callback URIs are usually specific endpoints to an application. Ideally, they shouldn't have any placeholders?
> Are we making these changes considering system apps?

Yes, this change is done for the system apps. Change is introduced in https://github.com/wso2/identity-apps/pull/1452 
For other apps, templated values are not accepted.